### PR TITLE
Type completion with tooltips 

### PR DESCRIPTION
**Pull request by AspenForester**
Resolves https://github.com/PowerShell/vscode-powershell/issues/2898
Adds tool tips for type completions, tests for completion are included.

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
hmm the Windows failures are different than the Windows failures...

File: test/PowerShellEditorServices.Test.Shared/Completion/CompleteTypeName.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                startLineNumber: 21,
+                startColumnNumber: 25,
+                startOffset: 0,
+                endLineNumber: 0,
+                endColumnNumber: 0,
+                endOffset: 0);
+
+        public static readonly CompletionDetails ExpectedCompletion =
+            CompletionDetails.Create(
+                "ArrayList",
```
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test.Shared/Completion/CompleteTypeName.cs:25]:
```suggestion
                "System.Collections.ArrayList",
```

**Comment by TylerLeonhardt** [comment]
So apparently `CompletionType.Namespace` is a thing... which would be `System.Collections` where as `CompletionType.Type` is `System.Collections.ArrayList` so you should also have a:

```csharp
case CompletionType.Namespace:
```

**Comment by TylerLeonhardt** [comment]
This test passes in PS7 but fails in Windows PowerShell

File: test/PowerShellEditorServices.Test.Shared/Completion/CompleteNamespace.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+namespace Microsoft.PowerShell.EditorServices.Test.Shared.Completion
+{
+    internal class CompleteNamespace
+    {
+        public static readonly ScriptRegion SourceDetails =
+            new ScriptRegion(
+                file: TestUtilities.NormalizePath("Completion/CompletionExamples.psm1"),
+                text: string.Empty,
+                startLineNumber: 21,
+                startColumnNumber: 25,
```
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test.Shared/Completion/CompleteNamespace.cs:17]: I think you want this to be like 15
```suggestion
                startColumnNumber: 15,
```

**Comment by AspenForester** [comment]
Unfortunately, I don't understand why the tests fail.  The two I created with @TylerLeonhardt's assistance seem to follow the model of the existing tests, but the completions results come back empty for the example provided for type completion.

**Review by rjmholt — APPROVED** [review]

**Comment by TylerLeonhardt** [comment]
A whole lot of nonsense for two lines of code... Thanks for putting up with the headache @AspenForester. Not sure why the tests don't work on Windows PowerShell but at least we'll have coverage for other PowerShell's and you confirmed it worked for Windows PowerShell manually...

I've opened https://github.com/PowerShell/PowerShellEditorServices/issues/1355 to track the test weirdness.

**Comment by TylerLeonhardt** [comment]
CI is green so I'm merging.

**Pull request merged**